sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -580,6 +580,12 @@ public FileBasedWriter(FileBasedWriteOperation<T> writeOperation) {
     protected abstract void prepareWrite(WritableByteChannel channel) throws Exception;
 
     /**
+     * Called after all calls to writeHeader, writeFooter, and write.
+     * If any resources opened in the write processes need to be closed, close them here.
```

Comment:
change "close" to "flush"?

As long as the `FileBasedWriter` does not "own" the channel (which it doesn't, because it doesn't create it), it should not close the channel.